doc/design/web.md

```diff
@@ -0,0 +1,141 @@
+# Web Interface design
+
+This design doc will talk about features and web pages needed to let users to manage cloud paddle jobs.
+
+## Feature List
+
+- Account Management
+    - Registration, send email to inform if registration suceeded
+    - Account Login/Logout
+    - Password changing, find back
+    - Download SSL keys
+- Jupiter Notebook
+    - Private Jupiter Notebook environment to run Python scripts
+    - Private workspace
+    - Submit job from Jupiter Notebook
+- Job Dashboard
+    - Job history and current running jobs
+    - Performance Monitoring
+    - Quota Monitoring
+- Datasets
+    - Public Dataset viewing
+    - Upload/Download private datasets
+    - Share datasets
+- Models
+    - Upload/Download models file
+    - Share/Publish Models
+- Paddle Board
+    - Training metrics visualization
+        - cost
+        - evaluator
+        - user-defined metrics
+- Serving
+    - Submit serving instances
+    - Deactivate serving
+    - Serving performance monitoring
+
+## Account Management
+
+I'll skip this section because it is a design that almost every website need.
+
+## Jupiter Notebook
+
+Start a ReplicaSet using image `docker.paddlepaddle.org/book` in kubernetes cluster and add an ingress endpoint when user first enters the notebook page.
+
+<img src="pictures/notebook.png" width="500px" align="center">
+
+Users can write program in python in the web page and save their programs, which will be saved at cloud storage. Users also can run script like below to submit a cluster training job:
+
+```python
+sess = paddle.framework.remote_session(
+    topology=block,
+    reader=reader,
+    jobname="cluster_job",
+    gpu=12
+)
+sess.run()
+```
+
+After this, there will be a job description and perfomance monitoring pages able to view at "Job Dashboard"
+
+## Job Dashboard
```

Comment:
Which program would serve this job dashboard?  As it is per-job, it seems that the master process of a job should serve it.  If so, it could be part of the PaddleBoard.

Reply:
No, job dashboard will list all jobs for the current user. Job dashboard is just one web page simply calls Kubernetes API to get the job list.